src/main/java/catbot/CatBotCommandPatterns.java

```diff
@@ -0,0 +1,63 @@
+package catbot;
+
+import catbot.internal.CommandPattern;
+import catbot.internal.CommandPatternGenerator;
+import catbot.internal.NamedParameterMap;
+
+import java.util.function.Consumer;
+
+public class CatBotCommandPatterns {
+
+    public static final Consumer<String> NO_DEFAULT = new Consumer<String>() {
+        @Override
+        public void accept(String s) {
+            assert false; //SHOULD NOT BE CALLED
+        }
+    };
+
+    //region Integer Pattern
+    public static CommandPatternGenerator<Integer> getIntegerPatternGenerator() {
+        return integerPatternGenerator;
+    }
+    private static final IntegerPatternGenerator integerPatternGenerator = new IntegerPatternGenerator();
+    private static class IntegerPatternGenerator implements CommandPatternGenerator<Integer> {
+
+        @Override
+        public CommandPattern<Integer> generate(Consumer<String> invalidInput) {
+            return new CommandPattern<Integer>() {
+                @Override
+                public void ifParsableElseDefault(String args, Consumer<Integer> consumer) {
+                    try {
+                        consumer.accept(Integer.parseInt(args));
+                    } catch (NumberFormatException nfe) {
+                        invalidInput.accept(args);
+                    }
+                }
+            };
+        }
+    }
+    //endregion
+
+    //region Slash Arguments Pattern
+    private static final SlashArgumentPatternGenerator slashPatternGenerator = new SlashArgumentPatternGenerator();
```

Comment:
Perhaps having shorter variable names may increase the readability of the code?

Reply:
Perhaps, but these are very infrequently used, so I thought it might have been better to make it very descriptive instead.
I'll think about it, but as of right now I'm still leaning towards my current convention, even if it results in excessively long variable names.